TestResultSummaryService/DataManager.js

```diff
@@ -49,6 +50,44 @@ class DataManager {
         return -1;
     }
 
+    async updateApplicationTests( data ) {
+        const { testName, testData, buildName, timestamp, _id, ...newData } = data;
+        if ( testData && testData.externalTestExtraData ) {
+            const buildInfo = Utils.getInfoFromBuildName(buildName);
+            if ( buildInfo ) {
+                const appDB = new ApplicationTestsDB();
+                const { dockerOS, appVersion, appName, javaVersionOutput } = testData.externalTestExtraData;
+                const { jdkVersion, jdkImpl, platform } = buildInfo;
+                let query = {
+                    testName,
+                    jdkVersion,
+                    jdkImpl,
+                    jdkPlatform: platform,
+                    dockerOS,
+                    appVersion,
+                    appName,
+                };
+                const update = {
+                    testId: _id,
+                    timestamp,
+                    javaVersionOutput,
+                    buildName,
+                    ...newData
+                }
+                const existingData = await appDB.getData(query).toArray();
+                if ( existingData && existingData.length > 0 ) {
+                    query = {
+                        timestamp: { $lt: timestamp },
+                        ...query
+                    }

```

Comment:
Instead of using another query, could we just check `if (timestamp > existingData[0].timestamp)`?

Reply:
This is not another query, this is the query for the update `appDB.update( query, { $set: update } )`. I can do a loop and check for `if (timestamp > existingData[0].timestamp)`  and do an update for each one. Or do you mean something else?

Reply:
We should only have one record, right? Why we need loop?

Reply:
updated